features/environment.py

```diff
@@ -134,17 +134,24 @@ def after_all(context):
 
         body_content = json.dumps({"passed": not context.failed})
         context.logger.info("Updating sauce job with %s" % body_content)
-
-        # If we need to go through a proxy server, see git history from August 2015 for the code to revert to
-        # See DEV-1109 in internal Jira for full discussion
-
-        connection = httplib.HTTPConnection("saucelabs.com")
-        connection.request('PUT', '/rest/v1/%s/jobs/%s' %
+        
+        # If a proxy is present then use it
+        # Otherwise connect directly to saucelabs
+        http_proxy = os.getenv('http_proxy', None)
+        if http_proxy:
+            if http_proxy.startswith("http://"):
+                http_proxy = http_proxy[7:]
+            connection = httplib.HTTPConnection(http_proxy)
+        else:
+            connection = httplib.HTTPConnection("saucelabs.com")
+
+        connection.request('PUT', 'https://saucelabs.com/rest/v1/%s/jobs/%s' %
```

Comment:
While it is hard to tell, this restores the file back to its long-standing form, but with one change: https instead of http for communicating with saucelabs.

This keeps the file consistent with all our other setups for communicating with sauce